hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/schema/HazelcastSchemaUtils.java

```diff
@@ -65,30 +68,33 @@ public static HazelcastSchema createRootSchema(List<TableResolver> tableResolver
 
             for (Table table : tables) {
                 HazelcastTable convertedTable = new HazelcastTable(
-                    table,
-                    createTableStatistic(table)
+                        table,
+                        createTableStatistic(table)
                 );
 
-                Map<String , HazelcastTable> schemaTableMap =
-                    tableMap.computeIfAbsent(table.getSchemaName(), (k) -> new HashMap<>());
+                if (table.getSchemaName() == null || table.getSchemaName().isEmpty()) {
```

Comment:
We cannot use empty schema names. A classical object hierarchy is `catalog -> schema -> object`. Every such an object should have a definitive name for several reasons:
1. They need to be displayed somehow in tools that will talk to us through JDBC
1. It should be possible to reach any object from any naming context. With empty schema name, some objects may become unreachable

Other vendors typically have a default schema name `PUBLIC` or alike.